main.js

```diff
@@ -0,0 +1,255 @@
+    return `${city.name} is in ${city.country} and has an area of ${city.areaInKilometers} square kilometers`;
+}
+
+console.log(getCityInformation({
+    name: 'Warsaw',
+    country: 'Poland',
+    areaInKilometers: 517
+}));
+
+// 9
+
+const john = {
+    firstName: 'John',
+    lastName: 'Smith'
+}
+
+function getUserCopy(user) {
+    return {
+        firstName: user.firstName,
+        lastName: user.lastName
+    }
+}
+
+const newUser = getUserCopy(john);
+console.log(newUser.firstName);
+console.log(newUser.lastName);
+console.log(newUser === john);
+
+// 10
+
+function getAnimalDescription(animal){
+    return `This ${animal.color} ${animal.name} has ${animal.legs} legs.`
+}
+
+// 11
+
+const rooms = {
+    first: {
+        description: 'This is the first room',
+        items: {
+            chair: 'The old chair looks comfortable',
+            lamp: 'This lamp looks ancient'
+        }
+    },
+    second: {
+        description: 'This is the second room',
+        items: {
+            couch: 'This couch looks like it would hurt your back',
+            table: 'On the table there is an unopened bottle of water'
+        }
+    }
+}
+
+// 12
+
+function addContact(contacts, name, phoneNumber) {
+    return {
+        ...contacts,
+        [name]: {
+            name: name,
+            phoneNumber: phoneNumber
+        }
+    };
+}
+
+const contacts = {};
+const contactsWithJohn = addContact(contacts, 'John', '123 456 789');
+
+console.log(contactsWithJohn.John.name);
+console.log(contactsWithJohn.John.phoneNumber);
+console.log(contacts === contactsWithJohn);
+console.log(contacts.John);
+
+// 13
+
+function findContact(contactList, name) {
+    if (contactList.hasOwnProperty(name)) {
+        return `Name: ${name}, Phone: ${contactList[name].phoneNumber}`;
+    }
+}
+
+const johnContactInfo = findContact(contactsWithJohn, 'John');
+console.log(johnContactInfo);
+
+// 14
+
+const userThree = {
+    firstName: 'Bob',
+    lastName: 'Ross',
+    age: 40,
+    greet() {
+        return "Hi, I'm Bob Ross!";
+    }
+}
+const objectExample = {
+    firstName: 'Kate',
+    lastName: 'Williams',
+    age: 40,
+    greet() {
+        return "Hi, I'm Kate!";
+    }
+}
+const car = {
+    make: 'Ferrari',
+    model: 'F40',
+    greet() {
+        return "<engine noise>";
+    }
+}
+
+function validateIfFirstNameisString(object) {
+    return typeof object.firstName === 'string';
+}
+
+function validateIfLastNameisString(object) {
+    return typeof object.lastName === 'string';
+}
+
+function validateIfAgeIsNumber(object) {
+    return typeof object.age === 'number';
+}
+
+function validateIfGreetIsFunction(object) {
+    return typeof object.greet === 'function';
+}
+
+function validateIfGreetContainsBothNames(object) {
+    return object.greet().includes(object.firstName)
+        && object.greet().includes(object.lastName);
+}
+
+function validateIfObjectIsAUser(object) {
+    const isFirstNameString = validateIfFirstNameisString(object);
```

Comment:
If a function is that simple and you are not reusing it anywhere, I would not create a function and just do
```
const isFirstNameString = typeof object.firstName === 'string';
```